src/ion/moves.rs

```diff
@@ -522,58 +502,38 @@ impl<'a, F: Function> Env<'a, F> {
                 // giving the allocation location for each label.
                 if !debug_labels.is_empty() {
                     // Do a binary search to find the start of any
-                    // relevant labels. Recall that we require
-                    // debug-label requests to be sorted by vreg then
-                    // by inst-range, with non-overlapping ranges, as
-                    // preconditions (which we verified above).
+                    // labels for this vreg. Recall that we require
+                    // debug-label requests to be sorted by vreg as a
+                    // precondition (which we verified above).
                     let start = debug_labels
-                        .binary_search_by(|&(label_vreg, _label_from, label_to, _label)| {
+                        .binary_search_by(|&(label_vreg, _label_from, _label_to, _label)| {
```

Comment:
Not strictly related to this PR, but I recently found out about the [`partition_point`](https://doc.rust-lang.org/nightly/std/primitive.slice.html#method.partition_point) method which may be more appropriate here.